modules/smartadserverBidAdapter.js

```diff
@@ -178,21 +177,27 @@ export const spec = {
       const videoMediaType = deepAccess(bid, 'mediaTypes.video');
       const bannerMediaType = deepAccess(bid, 'mediaTypes.banner');
       const isAdUnitContainingVideo = videoMediaType && (videoMediaType.context === 'instream' || videoMediaType.context === 'outstream');
+
       if (!isAdUnitContainingVideo && bannerMediaType) {
         payload.sizes = spec.adaptBannerSizes(bannerMediaType.sizes);
+        payload.bidfloor = bid.params.bidfloor || spec.getBidFloor(bid, adServerCurrency, BANNER);
         bidRequests.push(spec.createServerRequest(payload, bid.params.domain));
       } else if (isAdUnitContainingVideo && !bannerMediaType) {
         spec.fillPayloadForVideoBidRequest(payload, videoMediaType, bid.params.video);
+        payload.bidfloor = bid.params.bidfloor || spec.getBidFloor(bid, adServerCurrency, VIDEO);
         bidRequests.push(spec.createServerRequest(payload, bid.params.domain));
       } else if (isAdUnitContainingVideo && bannerMediaType) {
         // If there are video and banner media types in the ad unit, we clone the payload
         // to create a specific one for video.
         let videoPayload = deepClone(payload);
 
         spec.fillPayloadForVideoBidRequest(videoPayload, videoMediaType, bid.params.video);
+        videoPayload.bidfloor = bid.params.bidfloor || spec.getBidFloor(bid, adServerCurrency, VIDEO);
         bidRequests.push(spec.createServerRequest(videoPayload, bid.params.domain));
 
         payload.sizes = spec.adaptBannerSizes(bannerMediaType.sizes);
+        payload.bidfloor = bid.params.bidfloor || spec.getBidFloor(bid, adServerCurrency, BANNER);
+
```

Comment:
Can this be rewritten to something like the following so logic is not being repeated?

```
if (bannerMediaType) {
    if (isAdUnitContainingVideo) {
        // If there are video and banner media types in the ad unit, we clone the payload
        // to create a specific one for video.
        let videoPayload = deepClone(payload);

        spec.fillPayloadForVideoBidRequest(videoPayload, videoMediaType, bid.params.video);
        videoPayload.bidfloor = bid.params.bidfloor || spec.getBidFloor(bid, adServerCurrency, VIDEO);
        bidRequests.push(spec.createServerRequest(videoPayload, bid.params.domain));
    }
        
    payload.sizes = spec.adaptBannerSizes(bannerMediaType.sizes);
    payload.bidfloor = bid.params.bidfloor || spec.getBidFloor(bid, adServerCurrency, BANNER);
    bidRequests.push(spec.createServerRequest(payload, bid.params.domain));
} else if (isAdUnitContainingVideo && !bannerMediaType) {
    spec.fillPayloadForVideoBidRequest(payload, videoMediaType, bid.params.video);
    payload.bidfloor = bid.params.bidfloor || spec.getBidFloor(bid, adServerCurrency, VIDEO);
    bidRequests.push(spec.createServerRequest(payload, bid.params.domain));
} else {
    bidRequests.push({});
}
```